File: domain/src/main/java/io/prizy/domain/contest/ports/ContestSubscriptionRepository.java
package io.prizy.domain.contest.ports;

import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

import io.prizy.domain.contest.model.ContestSubscription;

/**
 * @author Nidhal Dogga
 * @created 4/24/2022 12:27 AM
 */

public interface ContestSubscriptionRepository {
  Collection<ContestSubscription> ofContest(UUID contestId);

  Collection<ContestSubscription> ofUser(UUID userId);

  Optional<ContestSubscription> subscriptionOfUser(UUID userId, UUID contestId);

  Boolean userSubscribedTo(UUID userId, UUID contestId);

  ContestSubscription create(ContestSubscription subscription);
}

File: domain/src/main/kotlin/io/prizy/domain/contest/ports/ContestSubscriptionRepository.kt
package io.prizy.domain.contest.ports

import io.prizy.domain.contest.model.ContestSubscription
import java.util.UUID

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 12:27 AM
 */

interface ContestSubscriptionRepository {
  suspend fun subscriptionsOf(contestId: UUID): Collection<ContestSubscription>
  suspend fun userSubscribedTo(userId: UUID, contestId: UUID): Boolean
  suspend fun create(subscription: ContestSubscription): ContestSubscription
}
